torchsynth/globals.py

```diff
@@ -31,9 +31,8 @@ def __init__(
         self.buffer_size = buffer_size
 
     def to(self, device: torch.device):
-        self.batch_size = self.batch_size.to(device)
-        self.sample_rate = self.sample_rate.to(device)
-        self.buffer_size = self.buffer_size.to(device)
+        # Only helpful to have sample rate on GPU, and as a float
```

Comment:
Interesting, any guesses why?

Reply:
So the difference between it being a plain float on the CPU and a float on the GPU was pretty small. But the really slow down happened when it was an int on the GPU. I think that was the same for the other attributes -- my thought (needs to be tested) is that mixing different types of tensors on the GPU is what causes slowdowns.